### PR TITLE
Switch to using dbus for dhcp lease info

### PR DESCRIPTION
**Pull request by ajeddeloh**
Use dbus for obtaining dhcp lease info instead of parsing files in /run/systemd/netif/leases/*. 
Depends on https://github.com/coreos/systemd/pull/50

### CODE REVIEW

File: internal/providers/azure/azure.go [hunk 1]
```diff
@@ -76,57 +76,68 @@ func getClient() retry.Client {
 	return client
 }
 
-func findLease() (*os.File, error) {
+/* godbus doesn't have a way to escape things. This will escape leading numbers, which is all we need */
+func escapeNumber(num int) string {
+	return "_3" + strconv.FormatUint(uint64(num), 10)
```
Comment (crawford) [inline internal/providers/azure/azure.go:81]: This only works where `num` is [0, 10), right?

Comment (ajeddeloh) [inline internal/providers/azure/azure.go:81]: Should work for all numbers, only leading numbers need to be escaped. E.g. 50 would become "_350"

Comment (crawford) [inline internal/providers/azure/azure.go:81]: Ah, gotcha.

File: internal/providers/azure/azure.go [hunk 2]
```diff
@@ -76,57 +76,68 @@ func getClient() retry.Client {
...
+		if !hasLease {
+			continue
 		}
-	}
 
-	if len(rawEndpoint) == 0 || len(rawEndpoint) != 8 {
-		return nil, fmt.Errorf("fabric endpoint not found in leases")
-	}
+		//then get the data from the lease, if we found it
+		leaseObjPath := dbus.ObjectPath("/org/freedesktop/network1/link/" + escapeNumber(link.Index) + "/lease")
```
Comment (crawford) [inline internal/providers/azure/azure.go]: Can you reuse the linkObjPath above and append "/lease"?

Comment (ajeddeloh) [inline internal/providers/azure/azure.go]: Fixed

**Comment by ajeddeloh** [comment]
Bump (thought I did this a day or two ago, guess I hadn't)

**Comment by cgonyeo** [comment]
Is this still something that should get in? I can review if so.

**Comment by ajeddeloh** [comment]
No, it relies on patches to systemd that were rejected upstream because they had other plans for a networkd dbus interface (see https://github.com/systemd/systemd/pull/4435 and https://github.com/systemd/systemd/pull/3528).

**Comment by cgonyeo** [comment]
Should this be closed then?

**Pull request closed without merging**